src/Raven.Client/ServerWide/Operations/CompareExchange.cs

```diff
@@ -15,34 +17,74 @@ public class RawClusterValueResult
     {
         public BlittableJsonReaderObject Value;
         public long Index;
+        public bool Successful;
     }
 
-    public class ClusterValueResult<T>
+    public class CmpXchgResult<T>
     {
         public T Value;
         public long Index;
+        public bool Successful;
+
+        public static CmpXchgResult<T> ParseFromBlittable(BlittableJsonReaderObject response, DocumentConventions conventions)
+        {
+            if (response.TryGet(nameof(RawClusterValueResult.Index), out long index) == false)
+                throw new InvalidDataException("Response is invalid.");
+
+            response.TryGet(nameof(RawClusterValueResult.Successful), out bool successful);
+            response.TryGet(nameof(RawClusterValueResult.Value), out BlittableJsonReaderObject raw);
+
+            T result;
+            object val = null;
+            raw?.TryGet("Object", out val);
+
+            if (val == null)
+            {
+                return new CmpXchgResult<T>
+                {
+                    Index = index,
+                    Value = default(T),
+                    Successful = successful
+                };
+            }
+            if (val is BlittableJsonReaderObject obj)
+            {
+                result = (T)EntityToBlittable.ConvertToEntity(typeof(T), "cluster-value", obj, conventions);
+            }
+            else
+            {
+                raw.TryGet("Object", out result);
+            }
+
+            return new CmpXchgResult<T>
+            {
+                Index = index,
+                Value = result,
+                Successful = successful
+            };
+        }
     }
 
-    public class GetClusterValue<T> : IServerOperation<ClusterValueResult<T>>
+    public class GetCompareExchangeOperation<T> : IOperation<CmpXchgResult<T>>
     {
         private readonly string _key;
 
-        public GetClusterValue(string key)
+        public GetCompareExchangeOperation(string key)
         {
             _key = key;
         }
 
-        public RavenCommand<ClusterValueResult<T>> GetCommand(DocumentConventions conventions, JsonOperationContext context)
+        public RavenCommand<CmpXchgResult<T>> GetCommand(IDocumentStore store, DocumentConventions conventions, JsonOperationContext context, HttpCache cache)
         {
-            return new GetClusterValueCommand(_key, conventions);
+            return new GetCompareExchangeValueCommand(_key, conventions);
         }
 
-        private class GetClusterValueCommand : RavenCommand<ClusterValueResult<T>>
+        private class GetCompareExchangeValueCommand : RavenCommand<CmpXchgResult<T>>
```

Comment:
Not sure if this is a typo or not. The operation is called GetCompareExchange, but the command also adds Value. I'm not sure which is a proper name.